imap_processing/tests/hit/test_hit_l1a.py

```diff
@@ -1,29 +1,264 @@
+            9: {"species": "Fe", "energy_idx": 0},
+        }
+        # Use the mod 10 value to determine the species and energy index
+        # for each science frame and add this information to the data frame
+        data["species"] = data["mod_10"].apply(
+            lambda row: species_energy[row]["species"].lower()
+            if row is not None
+            else None
+        )
+        data["energy_idx"] = data["mod_10"].apply(
+            lambda row: species_energy[row]["energy_idx"] if row is not None else None
+        )
+        data.drop(columns=["sectorates_by_mod_val", "mod_10"], inplace=True)
+        return data
+
+    def compare_data(expected_data, actual_data, skip):
+        # Compare the processed data to the validation data
+        for field in expected_data.columns:
+            if field not in [
+                "sc_tick",
+                "hdr_status_bits",
+                "species",
+                "energy_idx",
+            ]:
+                assert field in l1a_counts_data.data_vars.keys()
+            if field not in ignore:
+                for frame in range(expected_data.shape[0]):
+                    if field == "species":
+                        species = expected_data[field][frame]
+                        energy_idx = expected_data["energy_idx"][frame]
+                        assert np.array_equal(
+                            actual_data[f"{species}_counts_sectored"][frame][
+                                energy_idx
+                            ].data,
+                            expected_data["sectorates"][frame],
+                        )
+                    else:
+                        assert np.array_equal(
+                            actual_data[field][frame].data, expected_data[field][frame]
+                        )
+
+    rate_columns = {
+        "coinrates": "COINRATES_",
+        "bufrates": "BUFRATES_",
+        "l2fgrates": "L2FGRATES_",
+        "l2bgrates": "L2BGRATES_",
+        "l3fgrates": "L3FGRATES_",
+        "l3bgrates": "L3BGRATES_",
+        "penfgrates": "PENFGRATES_",
+        "penbgrates": "PENBGRATES_",
+        "sectorates": "SECTORATES_",
+        "l4fgrates": "L4FGRATES_",
+        "l4bgrates": "L4BGRATES_",
+        "ialirtrates": "IALIRTRATES_",
+        "sngrates_hg": "SNGRATES_HG_",
+        "sngrates_lg": "SNGRATES_LG_",
+    }
+
+    # Prepare validation data for comparison with processed data
+    validation_data.columns = validation_data.columns.str.strip()
+    validation_data = consolidate_rate_columns(validation_data, rate_columns)
+    validation_data = process_single_rates(validation_data)
+    validation_data = process_sectorates(validation_data)
+
+    # Fields to skip in comparison. CCSDS headers plus a few others that are not
+    # relevant to the comparison.
+    ignore = [
+        "version",
+        "type",
+        "sec_hdr_flg",
+        "pkt_apid",
+        "seq_flgs",
+        "src_seq_ctr",
+        "pkt_len",
+        "sc_tick",
```

Comment:
I am surprised that HIT doesn't want to carry `sc_tick`.

Reply:
They do, it's just that their validation data only has one `sc_tick` value per science frame (groups of 20 packets) and the dataset has values for all packets, so I'm skipping it for now. The `epoch` variable contains a converted sc_tick time per science frame and is the dimension used by all the rates data. `sc_tick` is only being used as a dimension for the ccsds headers which contain data for all the packets and therefore matches in shape